### PR TITLE
 [WIP][www] Sidebars à la reactjs.org (accordions, scrollspy)

### PR DESCRIPTION
**Pull request by fk**
…this time against master instead of v1 (ref. #6076 and https://github.com/fk/gatsby/pull/2).

### CODE REVIEW

**Comment by fk** [comment]
@m-allanson 

> I changed a Bound Action Creators to Actions.

D'oh, thanks for catching that! And I was so proud for catching the changed link and Tutorial Part 8! ;-)

> Is it possible to update the scroll position of the sidebar when changing pages in the mobile layout? The menu will open with the scroll position set to 0, instead of being scrolled to the currently selected page.

👍 Yeah for sure! We also want that for desktop, same issue there with the stuff further down.

> Would it be useful for the top level docs headers to link through to their own page for non-JS users?

Yeah, also wondered about that. Also I think we either need to throw in a `<h3>` again (this currently just uses a `<button>`) or update how we feed data to Algolia: 

image

File: www/src/components/sidebar/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+              borderRight: `1px solid ${colors.ui.light}`,
+              width: rhythm(10),
+              maxWidth: `none`,
+              backgroundColor: colors.ui.whisper,
+            },
+            [presets.Desktop]: {
+              width: rhythm(12),
+            },
+            [presets.Hd]: {
+              width: rhythm(14),
```
[the remark this answers is not in the document]
Comment (fk) [reply www/src/components/sidebar/sidebar.js]: 👍

**Comment by m-allanson** [comment]
@fk 👍 The selectors used by Algolia are configured over here https://github.com/algolia/docsearch-configs/blob/master/configs/gatsbyjs.json#L36.

**Comment by fk** [comment]
@m-allanson aceb421 brings back `<h3>` to the section titles.

I just chatted about this PR with @calcsam a few minutes ago. We decided to merge this as-is and follow up with issues tracking the remaining work.

**Comment by fk** [comment]
Can I get another review @calcsam? 🙏 😉

File: www/src/components/sidebar/section.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    const uid = `section_` + this.state.uid
+    const activeItemId = getActiveItem(section, location, activeItemHash)
+    const SectionTitleComponent = section.disableAccordions
+      ? Title
+      : ToggleSectionButton
+
+    let { isActive } = this.props
+    if (section.disableAccordions) {
+      isActive = true
+    }
```
Comment (calcsam) [inline www/src/components/sidebar/section.js]: don't modify props -- do something like:

`const isActive = this.props.isActive || section.disableAccordions`

**Comment by calcsam** [comment]
Did a quick scan and looks good!

File: www/src/components/sidebar/section.js (hunk 2 above)
Comment (fk) [reply www/src/components/sidebar/section.js]: Oh dang! 😁 Babysteppin'…

**Comment by fk** [comment]
> It's easy see how we could expand the Docs section to add in new Docs sections. How will this work on the tutorial section? Main question is -- how will subsequent tutorials be discoverable, since everything we put after the main tutorial series will be so far below the fold? Collapse everything by default? _Nest everything under a Introduction to Gatsby section that is opened by default on the /tutorial route, that can be fully collapsed with one click?_ Etc.

Finally getting through to answer this, sorry for taking so long @calcsam.

I'd split the current/main Tutorial contents into collapsible parts/groups and automatically expand the first—essentially what you suggest in your last sentence.
That should cover the need of people new to Gatsby and the Tutorial section in particular. If we implement storing the state of accordions in local storage or a cookie, people can move through the tutorial without having to manually navigate to the part of the tutorial where they stopped at.

**Comment by calcsam** [comment]
Awesome, thanks for the explanation @fk !

**Pull request merged**